File: BackEnd/apiFunctions/project.php
<?php
//*************************************************************************************************
// FileName : project.php
// FilePath : apiFunctions/
// Author   : Christian Marty
// Date		: 21.11.2023
// License  : MIT
// Website  : www.christian-marty.ch
//*************************************************************************************************
declare(strict_types=1);
global $database;
global $api;

if($api->isGet(Permission::Project_List))
{
    $query = <<< QUERY
        SELECT 
            ProjectNumber,
            Name,
            COALESCE(Description, '') AS Description
        FROM project
    QUERY;
	$result = $database->query($query);
    \Error\checkErrorAndExit($result);

	foreach($result as $item) {
		$item->ItemCode = \Numbering\parser(\Numbering\Category::Project, $item->ProjectNumber);
	}

	$api->returnData($result);
}

File: API/project.py
# *************************************************************************************************
# FileName : project.py
# Author   : Christian Marty
# Date     : 16.02.2024
# License  : MIT
# Website  : www.christian-marty.ch
# *************************************************************************************************

class Project(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance

    def list(self):
        data = self._eerp.get('project')
        return data
